tests/experiment/test_procedure.py

```diff
@@ -88,8 +93,18 @@ class TestProcedure(Procedure):
         @property
         def prop(self):
             return self.x
+
     p = TestProcedure()
     with pytest.raises(AttributeError):
         _ = p.prop  # AttributeError
     p.x = 5
     assert p.prop == 5
+
+
+@pytest.mark.parametrize("header, units", (
+        ("x (m)", ureg.m),
+        ("x (m/s)", ureg.m / ureg.s),
+        ("x (V/(m*s))", ureg.V / ureg.m / ureg.s),
+))
+def test_csv_formatter_parse_columns(header, units):
+    assert Procedure.parse_columns([header])[header] == ureg.Quantity(1, units)
```

Comment:
I don't know, whether you plan to do it: You can also test, that wrong inputs raise an error, for example
~~~python
with pytest.raises(DivideByZeroError):
    1/0
~~~
Will pass the test. The test will fail, if the exception is not raised!

A "match" parameter can verify, that a certain string is part of the error message.

Reply:
Just added two tests, one for valid headers with no units and one for invalid units in header.

Reply:
Thanks